### ISSUE TITLE
Feature Request: Pass Full Scroll Buffer To Alacritty When Single Pane Tab

### ISSUE THREAD

**Issue by sstubbs** [zellij-org/zellij#982]
Firstly thank you for this amazing project. 

I've been trying out Zellij in Alacritty which works really well so far. One of the features of Alacritty I like is using VI mode to find and copy things in logs of a running program while I'm working on it. If I have a tab with a single pane in it would be great if the full scroll buffer could be passed back to Alacritty rather than the scroll being managed by zellij.

I realise there is probably not a clean generic way of doing this though so not sure if it's possible but thought I would add it anyway so it could considered in future design. I use a window manager when using Linux but on Mac it's not ideal to open up multiple alacritty windows if I know something will be running that I need to search through and generally I would like to use Zellij on Linux and Mac so things are the same.

**Comment by imsnif** [comment]
Hey, happy you like Zellij!

This is an interesting idea and I'd totally be open to finding the user story to do this - I think that's the biggest challenge here.

The issue is that for this to work we'll have to (temporarily?) get rid of the tab bar (we can keep the status bar since it's on the bottom - but it might seem weird UX-wise?)

So this sounds like it will be a separate "mode". Kind of like full-screen (`ctrl+p` + `f`) but without part of the UI. But then, what happens when you toggle it on/off? You lose all the scroll history in the terminal emulator. We can probably dump it back manually every time, but if it's big there might be performance issues.

So, what I'm getting at: I know how to do this tech-wise, I'm not 100% sure how to do this UX-wise with all the limitations. Any thoughts?

**Comment by sstubbs** [comment]
Yes it certainly is a challenge from a UX point of view. Excited that it's technically possible though. What about the other way around? Pass the buffer to another program from a pane. So for example there would be a Zellij shortcut that opens the pane buffer in Helix or VIM within that pane.

I feel like the combination of Alacriity, Zellij and Helix pretty much are what I've been dreaming about for years. Really really excited about all 3 of these projects.

**Comment by imsnif** [comment]
> What about the other way around? Pass the buffer to another program from a pane. So for example there would be a Zellij shortcut that opens the pane buffer in Helix or VIM within that pane.

Yes! I've been wanting to do this for a long time. I just haven't gotten to it yet. It shouldn't be incredibly complicated. We already know how to open your default editor in a new pane. Need to find out how to pass it a text buffer in a generic way and then serialize the scroll buffer to it.

Is this something you'd be interested in working on?

**Comment by sstubbs** [comment]
I would love to work on this. Will pick it up in about a month when I have some free time. Will need a bit of guidance where to start though.

**Comment by imsnif** [comment]
Awesome. Feel free to give me a ping and I'd be happy to guide you through it.

I was thinking about this btw - we could probably get away with dumping the scrollback to a temporary folder (stripping colors and styling and such) and then pointing the text editor at it. Even easier.
